src/main/java/org/folio/md/validator/service/impl/UniquePermissionForEndpointValidator.java

```diff
@@ -0,0 +1,41 @@
+package org.folio.md.validator.service.impl;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.folio.md.validator.model.RoutingEntry;
+import org.folio.md.validator.model.ValidationContext;
+import org.folio.md.validator.service.Validator;
+
+public class UniquePermissionForEndpointValidator implements Validator {
+
+  @Override
+  public void validate(ValidationContext ctx) {
+    var moduleDescriptor = ctx.getModuleDescriptor();
+    moduleDescriptor.getProvides().stream()
```

Comment:
possible NPE

Reply:
moduleDescriptor cannot be null as well [as provides](https://github.com/folio-org/folio-module-descriptor-validator/blob/4b0fae0140b1f33b4ed90089dbc965a58bf3f0af/src/main/java/org/folio/md/validator/model/ModuleDescriptor.java#L22)